src/ui/customBreakpointsUI.ts

```diff
@@ -125,25 +180,68 @@ class BreakpointsDataProvider implements vscode.TreeDataProvider<Breakpoint | Ca
       breakpoint.checkboxState = state;
 
       if (breakpoint instanceof Category) {
-        for (const child of breakpoint.children) {
+        for (const child of this.getChildren(breakpoint) as XHRBreakpoint[]) {
           if (child.checkboxState !== state) {
             child.checkboxState = state;
           }
         }
-      } else {
-        if (!enabled && breakpoint.parent.checked) {
-          breakpoint.parent.checkboxState = state;
-        } else if (enabled && breakpoint.parent.children.every(c => c.checked)) {
-          breakpoint.parent.checkboxState = state;
+      } else if (breakpoint instanceof Breakpoint || breakpoint instanceof XHRBreakpoint) {
+        const parent = this.getParent(breakpoint) as Category;
+        if (!enabled && parent.checked) {
+          parent.checkboxState = state;
+        } else if (
+          enabled &&
+          (this.getChildren(parent) as XHRBreakpoint[]).every(
```

Comment:
If we add `parent` and `checked` on the XHRBreakpoint then we can avoid this :)

Reply:
`XHRBreakpoint` already has `checked`. What would be avoiding?